src/Chapter3.hs

```diff
@@ -907,9 +998,21 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+newtype Gold = Gold Int deriving Show
+
 class Append a where
     append :: a -> a -> a
 
+instance Append Gold where
+    append (Gold x) (Gold y) = Gold (x + y)
+
+instance Append [a] where
+    append = (++)
+
+instance Append a => Append (Maybe a) where
+    append :: Maybe a -> Maybe a -> Maybe a
+    append (Just x) (Just y) = Just (append x y)
+    append _ _ = Nothing
```

Comment:
Here, even if one of the arguments is `Just` it will return `Nothing`, though more correct to be to return the `Just`.